src/intake_sqlite/sqlite_src.py

```diff
@@ -0,0 +1,157 @@
+"""SQLite Intake driver classes."""
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from typing import Any
+
+import fsspec
+from intake_sql import SQLSource, SQLSourceAutoPartition, SQLSourceManualPartition
+
+import intake_sqlite
+
+logger = logging.getLogger(__name__)
+
+SQLITE_SUFFIXES = (".db", ".sqlite")
+
+
+__all__ = [
+    "SQLiteSource",
+    "SQLiteSourceAutoPartition",
+    "SQLiteSourceManualPartition",
+    "urlpath_to_sqliteurl",
+]
+
+
+class SQLiteSource(SQLSource):  # type: ignore
+    """Read the full results of an SQL query into a dataframe.
+
+    Args:
+        urlpath: A local path or :mod:`fsspec` readable URL pointing to a SQLite
+            database.
+        sql_expr: Query expression to pass to the SQLite database backend.
+        sql_kwargs: Additional arguments to pass in to :func:`pandas.read_sql`.
+        metadata: Arbitrary metadata dictionary associated with the data source.
+        open_kwargs: Additional arguments to pass to :func:`fsspec.open_local`
+
+    """
+
+    name = "sqlite"
+    version = intake_sqlite.__version__
+    container = "dataframe"
+    partition_access = True
+
+    def __init__(
+        self,
+        urlpath: str,
+        sql_expr: str,
+        sql_kwargs: dict[str, Any] = {},
+        metadata: dict[str, Any] = {},
+        open_kwargs: dict[str, Any] = {},
+    ):
+        """Initialize the class, transforming remote URL path to a local file path."""
+        super().__init__(
+            uri=urlpath_to_sqliteurl(urlpath, open_kwargs=open_kwargs),
+            sql_expr=sql_expr,
+            sql_kwargs=sql_kwargs,
+            metadata=metadata,
+        )
+
+
+class SQLiteSourceAutoPartition(SQLSourceAutoPartition):  # type: ignore
+    """SQLite Table reader with automatic partitioning.
```

Comment:
How do you determine the size of the partition?

Reply:
I wasn't totally clear on that either. I was just leaving it to `intake-sql` but I'll have to figure it out a bit more to write a test for this class.